pkg/util/timeutil/time_zone_util.go

```diff
@@ -68,3 +73,44 @@ func ParseFixedOffsetTimeZone(location string) (offset int, origRepr string, suc
 	}
 	return offset, strings.TrimSuffix(strings.TrimPrefix(origRepr, "("), ")"), true
 }
+
+// TimeZoneOffsetStringConversion converts a time string to offset seconds
+// Supported time zone strings :- GMT/UTC±[00:00:00 - 15:59:59], GMT/UTC±[0-16]
+// Unsupported time zone strings :- GMT/UTC±16:00 to upper, GMT/UTC±6.5
+// (case insensitive)
+func TimeZoneOffsetStringConversion(s string) (offset int64, ok bool) {
+	if !(re.MatchString(s)) {
+		return 0, false
+	}
+	prefix := string(re.FindStringSubmatch(s)[0][3])
```

Comment:
nit: going to re-use the result of `re.FindStringSubmatch` for you here, and also add a test for whitespaces.